core/src/main/java/com/novoda/merlin/RxCallbacksManager.java

```diff
@@ -0,0 +1,25 @@
+package com.novoda.merlin;
+
+import rx.Observable;
+import rx.subjects.BehaviorSubject;
+
+public class RxCallbacksManager {
+
+    private final BehaviorSubject<Boolean> merlinRxConnectionStatusSubject;
```

Comment:
I would usually agree with you since [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). In this case though since we are building a public API and it is not a big change in code I would go for the enum. It will allow for non breaking changes with regards to the API.